src/AppInstallerCLITests/PredefinedInstalledSource.cpp

```diff
@@ -400,3 +402,89 @@ TEST_CASE("PredefinedInstalledSource_Search", "[installed][list]")
 
     REQUIRE_FALSE(results.Matches.empty());
 }
+
+std::string GetDatabaseIdentifier(const std::shared_ptr<Repository::ISource>& source)
+{
+    return reinterpret_cast<SQLiteIndexSource*>(source->CastTo(ISourceType::SQLiteIndexSource))->GetIndex().GetDatabaseIdentifier();
+}
+
+TEST_CASE("PredefinedInstalledSource_Create_Cached", "[installed][list][installed-cache]")
+{
+    auto source1 = CreatePredefinedInstalledSource();
+    auto source2 = CreatePredefinedInstalledSource();
+
+    // Ensure the same identifier (which should mean the cache was not updated)
+    REQUIRE(
+        GetDatabaseIdentifier(source1)
+        ==
+        GetDatabaseIdentifier(source2)
+    );
+
+    // Get all packages
+    auto result1 = source1->Search({});
+    REQUIRE(!result1.Matches.empty());
+
+    for (const auto& match : result1.Matches)

```

Comment:
Could/should we check that there are no packages in result2 that aren't in result1?